Sieve_of_Eratosthenes/SieveOfEratosthenes.c

```diff
@@ -0,0 +1,34 @@
+#include <stdio.h>
+#include <math.h>     // for using square root 
+#include <stdbool.h>  // to include bool datatype 
+
+void sieveOfEratosthenes(int n){
+	bool numbers[n+1];
+	int i, sqrt_n = sqrt(n)+1, j;
+	
+	for (i = 0; i < n+1; i++){
+		numbers[i] = true;      // assigning every number "true" 
+	}
+
+	for (i = 2; i < sqrt_n; i++){
+		if (numbers[i]){
+			for (j = 2*i; j < n+1; j = j+i){
+				numbers[j] = false;
+			}
+		}
+	}
+
+	for (j=2; j < n; j++){
+		if (numbers[j]){
+			printf("%d ",j);
```

Comment:
Oh yes, I missed that. Indeed it will be `bool*` and the loop would start from 2.

Let me make the changes - 
```C
void sieveOfEratosthenes(int n, bool *primes) {
    /*
        Allocate memory to primes and store results in it.
    */
}

int main() {
    ...
    bool *primes;
    sieveOfEratosthenes(n, primes);
    for (i = 2 ; i < n+1 ; i ++) {
        if (primes[i])
            printf("%d\n", primes[i]);
    }
}
```